src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -256,31 +170,32 @@ class LspConnectionManager {
     return [startLineStyle, middleLinesStyle, endLineStyle]
   }
 
-  _setEditorBlockStyle(schema: CompositionSelection = this._session) {
-    const compositionBlock = this._getCompositionBlockLines(
-      this._model.getValue()
-    )
-
-    const removeAllStyles = !compositionBlock || schema.composition.diverged
+  _setEditorBlockStyle(range: LspRange | null) {
+    const removeAllStyles = range == null
 
     this._compositionStyle = this._editor.deltaDecorations(
       this._compositionStyle,
       removeAllStyles
         ? []
-        : this._compositionSyncStyle(
-            compositionBlock?.startLine,
-            compositionBlock?.endLine,
-            schema.composition.synced
-          )
+        : this._compositionSyncStyle(range?.start.line, range.end.line)
     )
   }
 
+  _isNewScript(
+    schema: CompositionSelection,
+    previousState: CompositionSelection
+  ): boolean {
+    return previousState.bucket != null && schema.bucket == null
```

Comment:
Purposefully playing a little loose here, since the source of this `bucket` is plenty of upstream, interconnected code which manages the session state. So I want to handle cases where we go from "user see nothing" (undefined or null), to user sees a bucket.